src/TuxArchitectStores.js

```diff
@@ -65,31 +65,53 @@ ArchitectChain.prototype.itOutputs = function (output) {
 //_getRegisterId FUNCTION: internal function,for getting the store __registerId__ from either the passed in store or the store mapped to the passed in input
 //@param inputOrStore STRING: input string to be checked against the architecture OBJECT for the corresponding store [ALTERNATE OBJECT: if a store is passed in then lookup __registerId__ on the passed in store]
 ArchitectChain.prototype._getRegisterId = function (inputOrStore) {
-  var storeToWaitFor, storeRegistrationId;
+  var storeOrStoresToWaitFor, storeRegistrationIdorIds;
   //if inputOrStore is a string
   if (typeof inputOrStore === 'string') {
-    //lookup corresponding store in the architecture object
-    storeToWaitFor = architecture[inputOrStore];
+    //lookup corresponding store or stores in the architecture object
+    storeOrStoresToWaitFor = architecture[inputOrStore];
     //if no store maps to this input throw error
-    invariant(storeToWaitFor, 'store is waiting for an input: "%s" that no store outputs.  If this store needs no inputs than only call "itOutputs" method.', inputOrStore);
+    invariant(storeOrStoresToWaitFor, 'store is waiting for an input: "%s" that no store outputs.  If this store needs no inputs than only call "itOutputs" method.', inputOrStore);
   } else {
     //if inputOrStore is a store use that
-    storeToWaitFor = inputOrStore;
+    storeOrStoresToWaitFor = inputOrStore;
   }
-  //return __registerId__ from storeToWaitFor
-  storeRegistrationId = storeToWaitFor.__registerId__;
-  //if store does not have __registerId__ than it has not been registered to the dispatcher so throw an error
-  invariant(storeRegistrationId, 'store is waiting for a store that has not been registered to any actions.');
-  return storeRegistrationId;
+  //return __register__id from storeOrStoresToWaitFor
+  storeRegistrationIdorIds = this._determineOneOrMultipleRegisterIds(storeOrStoresToWaitFor);
+  return storeRegistrationIdorIds;
 };
 
+ArchitectChain.prototype._determineOneOrMultipleRegisterIds = function(storeOrStoresToWaitFor) {
+  //if storeOrStoresToWaitFor is an array holding multiple stores
+  if (Array.isArray(storeOrStoresToWaitFor)) {
+    var storeRegistrationIds = [];
+    //grab the __registerId__ from each individual store and return them as an array
+    for (var i = 0; i < storeOrStoresToWaitFor.length; i++) {
+      storeRegistrationIds.push(storeOrStoresToWaitFor[i].__registerId__);
```

Comment:
Need to do invariant checking here as well.  Before pushing in each id, check each one and throw an error if it is not defined as we do on line 97.  You can store the error message in a string so you dont have to repeat it.